doc/nrf/glossary.rst

```diff
@@ -410,6 +410,11 @@ Glossary
    Noise Factor (NF)
       The relation of the :term:`Signal-to-Noise Ratio (SNR)` in the device input to the SNR in the device output.
 
+   Non-Secure Processing Environment (NSPE)
+      A concept related to the Security by Separation architecture principle, which affects both hardware and firmware.
+      In the Security by Separation scenario, NSPE refers to the firmware components located in the non-secure partition, such as main application, while all the security components reside in the :term:`Secure Processing Environment (SPE)`.
```

Comment:
```suggestion
      In the Security by Separation scenario, NSPE refers to the firmware components located in the non-secure partition, such as main user application, while all the security components reside in the :term:`Secure Processing Environment (SPE)`.
```

Does this correspond with the term used elsewhere in this PR?

Reply:
Rephrased both definitions in the glossary and added CMSE term.